goalie/function_data.py

```diff
@@ -80,12 +81,16 @@ def _data_by_label(self):
         layout: indexed first by subinterval and then by export.
         """
         tp = self.time_partition
+        _labels = []
+        for field_type in tp.field_types:
+            _labels.extend(self._label_dict[field_type])
+
         return AttrDict(
             {
                 label: AttrDict(
                     {field: self._data_by_field[field][label] for field in tp.fields}
                 )
-                for label in self.labels
+                for label in _labels
```

Comment:
Instead of duplicating this, why not just implement it once in the `__init__` method of the base class? The `labels` attribute that is computed there is never used with these changes. Why not just overwrite that definition with this new one? It works for me when I do that locally.

Reply:
Also, a list comprehension (rather than `extend`) would be more in line with the style used in Goalie.

Reply:
done in commit [6679c33](https://github.com/pyroteus/goalie/pull/149/commits/6679c33fe8b6b02d3f2607b172194e137d29f3d5)

I originally used `extend` to avoid the doubly nested list and having to add a `[0]` at the end of the list comprehension which seems a bit less readable. Hope it works now. The individual functions should be back to how they were before commit [c56c4af](https://github.com/pyroteus/goalie/pull/149/commits/c56c4afc338b5d05ce096a54d733d974c6ac7d1e).